src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -331,9 +331,10 @@ private bool IsString(int typeIndex)
 
         private static byte[] EncodeStringName()
         {
-            var type = typeof(string);
-            var name = type.AssemblyQualifiedName;
-            int length = name.IndexOf(", Version=");
+            // ResourceWriter always write the resources with legacy core library name (i.e. mscorlib).
+            // Ensure that when we read the type name, we update the library name to be the same.
+            var name = "System.String, mscorlib";
```

Comment:
Nit: Could this be a constant? e.g.:

```csharp
const string Name = "System.String, mscorlib";
```

with capitalized `Name` to match the corefx coding style for constants.